src/secp256k1/src/cpp/Scalar.cpp

```diff
@@ -268,27 +270,43 @@ unsigned char* Scalar::deserialize(unsigned char* buffer) {
 }
 
 std::string Scalar::GetHex() const {
-    unsigned char buffer[32];
-    serialize(buffer);
+    std::array<unsigned char, 32> buffer;
+    secp256k1_scalar_get_b32(buffer.data(), reinterpret_cast<const secp256k1_scalar *>(value_));
 
     std::stringstream ss;
     ss << std::hex;
-    for (int i = 0; i < 32; ++i) {
-        ss << buffer[i] / 16;
-        ss << buffer[i] % 16;
+    for (const auto b : buffer) {
+        ss << (b >> 4);
+        ss << (b & 0xF);
     }
 
     return ss.str();
 }
 
 void Scalar::SetHex(const std::string& str) {
-    unsigned char buffer[32];
+    if (str.size() != 64) {
+        throw "Scalar: decoding invalid length";
+    }
+
+    std::array<unsigned char, 32> buffer;
+
+    for (std::size_t i = 0; i < buffer.size(); i++) {
+        auto hexs = str.substr(2 * i, 2).c_str();
```

Comment:
This is a bug. `substr` return a new `std::string` that mean it will destroy immediately once a call to `c_str` is completed.